aten/src/ATen/native/mps/operations/LossOps.mm

```diff
@@ -537,7 +551,9 @@ static void nllnd_loss_forward_impl(Tensor& output,
           mpsGraphBatchSizeTensor = [mpsGraph reductionSumWithTensor:mpsSelectOneTensor
                                                                 axes:nil
                                                                 name:@"batchSizeReductionTensor"];
-          mpsGraphReducedTensor = divisionNoNaN(mpsGraph, mpsGraphReducedTensor, mpsGraphBatchSizeTensor);
+          mpsGraphReducedTensor = [mpsGraph divisionWithPrimaryTensor:mpsGraphReducedTensor
+                                                      secondaryTensor:mpsGraphBatchSizeTensor
+                                                                 name:@"divisionTensor"];
```

Comment:
So, why this is safe/needed?

Reply:
This is partially what this PR is fixing. It is needed in the case where `weight` elems are zero. Without it https://github.com/pytorch/pytorch/blob/39a61795e3ee41eff4dfe76da14b2535cf47429b/test/test_nn.py#L11599 will fail.